python/feathub/metric_stores/metric.py

```diff
@@ -53,17 +57,29 @@ def get_tags(self) -> OrderedDict:
             ],
         )
 
-    @abstractmethod
-    def get_transform_functions(self) -> Sequence[Callable[[str], Transformation]]:
+    def get_transform_functions_and_output_types(
+        self,
+    ) -> Sequence[
+        Tuple[Callable[[str], Transformation], Callable[[DType], Optional[DType]]]
+    ]:
         """
         Gets the functions to generate transformations used to aggregate and compute the
-        metric value. The function has the following parameter:
+        metric value, and the functions to derive the data type of the output metric
+        value.
 
+        The function to generate transformations has the following parameter:
         - feature_name: Name of a variable containing the intermediate metric value
                         generated from previously defined transformation. For the first
                         transformation, its value is the host feature name.
+
+        The function to derive data types has the following parameter:
+        - dtype: The data type of the input feature or the intermediate metric value
+                 generated from previously defined transformation.
         """
-        pass
+        return [(x, lambda dtype: None) for x in self._get_transform_functions()]
```

Comment:
Only the PythonUdfTransform will return an output dtype. It may be more intuitive to allow PythonUdfTransform to derive the output dtype from the input dtype in the long term. 

How about adding a TODO?